File: ShiftSync.Web/src/types/shiftsync/attendance.ts
export interface Attendance {
  id: number;
  userId: string;
  userShiftId: number;
  checkInTime: string;
  checkInLatitude: number;
  checkInLongitude: number;
  checkOutTime?: string;
  checkOutLatitude?: number;
  checkOutLongitude?: number;
  userFullName?: string;
  shiftName?: string;
}

export interface AttendanceFilter {
  pageNumber?: number;
  pageSize?: number;
  userId?: string;
  userShiftId?: number;
  dateFrom?: string;
  dateTo?: string;
}

File: ShiftSync.API/Models/Entities/Attendance.cs
namespace ShiftSync.API.Models.Entities;

public class Attendance
{
    public int Id { get; set; }

    public string UserId { get; set; } = string.Empty;
    public ApplicationUser User { get; set; } = null!;

    public int UserShiftId { get; set; }
    public UserShift UserShift { get; set; } = null!;

    public DateTime CheckInTime { get; set; }
    public double CheckInLatitude { get; set; }
    public double CheckInLongitude { get; set; }

    public DateTime? CheckOutTime { get; set; }
    public double? CheckOutLatitude { get; set; }
    public double? CheckOutLongitude { get; set; }

    public ICollection<AttendanceBreak> Breaks { get; set; } = new List<AttendanceBreak>();
}
